src/test/javascript/spec/component/tutorial-groups/crud/edit-tutorial-group.component.spec.ts

```diff
@@ -0,0 +1,151 @@
+        tutorialGroupOfResponse.title = 'test';
+
+        const response: HttpResponse<TutorialGroup> = new HttpResponse({
+            body: tutorialGroupOfResponse,
+            status: 200,
+        });
+
+        const findByIdStub = jest.spyOn(tutorialGroupService, 'getOneOfCourse').mockReturnValue(of(response));
+        const tutorialGroupFormStubComponent: TutorialGroupFormStubComponent = editTutorialGroupComponentFixture.debugElement.query(
+            By.directive(TutorialGroupFormStubComponent),
+        ).componentInstance;
+
+        editTutorialGroupComponentFixture.detectChanges();
+        expect(editTutorialGroupComponent.tutorialGroup).toEqual(tutorialGroupOfResponse);
+        expect(findByIdStub).toHaveBeenCalledWith(1, 1);
+        expect(editTutorialGroupComponent.formData.title).toEqual(tutorialGroupOfResponse.title);
+        expect(tutorialGroupFormStubComponent.formData).toEqual(editTutorialGroupComponent.formData);
+    });
+
+    it('should send PUT request upon form submission and navigate', () => {
+        const router: Router = TestBed.inject(Router);
+        const tutorialGroupService = TestBed.inject(TutorialGroupsService);
+        const exampleTeachingAssistant = new User();
+        exampleTeachingAssistant.login = 'testLogin';
+
+        const tutorialGroupInDatabase: TutorialGroup = new TutorialGroup();
+        tutorialGroupInDatabase.id = 1;
+        tutorialGroupInDatabase.title = 'test';
+        tutorialGroupInDatabase.teachingAssistant = exampleTeachingAssistant;
+
+        const findByIdResponse: HttpResponse<TutorialGroup> = new HttpResponse({
+            body: tutorialGroupInDatabase,
+            status: 200,
+        });
+        const findByIdStub = jest.spyOn(tutorialGroupService, 'getOneOfCourse').mockReturnValue(of(findByIdResponse));
+
+        editTutorialGroupComponentFixture.detectChanges();
+        expect(findByIdStub).toHaveBeenCalledWith(1, 1);
+        expect(editTutorialGroupComponent.tutorialGroup).toEqual(tutorialGroupInDatabase);
+
+        const changedTutorialGroup: TutorialGroup = {
+            ...tutorialGroupInDatabase,
+            title: 'Changed',
+        };
+
+        const updateResponse: HttpResponse<TutorialGroup> = new HttpResponse({
+            body: changedTutorialGroup,
+            status: 200,
+        });
+
+        const updatedStub = jest.spyOn(tutorialGroupService, 'update').mockReturnValue(of(updateResponse));
+        const navigateSpy = jest.spyOn(router, 'navigate');
+
+        const tutorialGroupForm: TutorialGroupFormStubComponent = editTutorialGroupComponentFixture.debugElement.query(
+            By.directive(TutorialGroupFormStubComponent),
+        ).componentInstance;
+
+        tutorialGroupForm.formSubmitted.emit({
+            title: changedTutorialGroup.title,
+            teachingAssistant: exampleTeachingAssistant,
+        });
+
+        expect(updatedStub).toHaveBeenCalledOnce();
+        expect(navigateSpy).toHaveBeenCalledOnce();
```

Comment:
Missing `toHaveBeenCalledWith()`

Reply:
Not needed imo as it is enough to know that navigation works.